src/test/java/com/redhat/devtools/lsp4ij/mock/MockLanguageServerDefinition.java

```diff
@@ -25,24 +27,36 @@ public class MockLanguageServerDefinition extends LanguageServerDefinition imple
 
     private static final String SERVER_ID = "mock-server-id";
 
+    private final boolean clientConfigurable;
     private final ClientConfigurationSettings clientConfigurationSettings = new ClientConfigurationSettings();
 
-    public MockLanguageServerDefinition() {
-        this(SERVER_ID);
+    public MockLanguageServerDefinition(boolean clientConfigurable) {
+        this(SERVER_ID, clientConfigurable);
     }
 
-    public MockLanguageServerDefinition(String serverId) {
+    public MockLanguageServerDefinition(@NotNull String serverId) {
+        this(serverId, false);
+    }
+
+    private MockLanguageServerDefinition(@NotNull String serverId, boolean clientConfigurable) {
         super(serverId, "name", null, true, 5, true);
+        this.clientConfigurable = clientConfigurable;
```

Comment:
This is where it's stored so the mock language server definition can behave accordingly.